src/data/matrix/basic.lean

```diff
@@ -1480,6 +1480,20 @@ instance [fintype n] [semiring α] [star_ring α] : star_ring (matrix n n α) :=
 lemma star_mul [fintype n] [non_unital_semiring α] [star_ring α] (M N : matrix n n α) :
   star (M ⬝ N) = star N ⬝ star M := conj_transpose_mul _ _
 
+lemma star_mul_vec [fintype n] [semiring α] [star_ring α] (M : matrix n n α) (v : n → α) :
+  star (M.mul_vec v) = (star M).vec_mul (star v) :=
+begin
+  ext i,
+  calc star (M.mul_vec v) i = star (M i ⬝ᵥ v) :
+    by simp only [mul_vec, pi.star_apply]
+  ... = star v ⬝ᵥ star (λ j, M i j) :
+    by rw [← star_dot_product_star]
+  ... = star v ⬝ᵥ λ (i_1 : n), Mᴴ i_1 i :
+    by simp only [conj_transpose_apply, star]
+  ... = (star M).vec_mul (star v) i :
+    by simp only [vec_mul, (star_apply _ _ _).symm, conj_transpose_apply]
+end
+
```

Comment:
I've split this out into #14248, since I was able to golf the proof down to 1 line and generalize the typeclass arguments.

Reply:
Oh, wow, that's great!

Reply:
A good strategy for proving trivial things about `vec_mul` and `mul_vec` is just to unfold everything, and then see if what you end up with is something you can already prove about `dot_product`.

Reply:
I pushed a merge commit that removed this lemma